app/Libraries/Search/BeatmapsetSearchRequestParams.php

```diff
@@ -134,6 +136,11 @@ public static function getAvailableFilters()
         return compact('extras', 'general', 'genres', 'languages', 'modes', 'played', 'ranks', 'statuses');
     }
 
+    public function isLoginRequired(): bool
+    {
+        return present($this->requestQuery);
+    }
```

Comment:
Interestingly, this doesn't actually affect `shouldReturnEmptyResponse` for this class :upside_down_face: